### ISSUE TITLE
Add an option to profile to see links to API objects on pages and search results

### ISSUE THREAD

**Issue by mlissner** [freelawproject/courtlistener#419]
For our API users, having a link from a opinion to the correct place in the API would be killer. Just for debugging you could click a link from an opinion page and go straight to the Opinion, Opinion Cluster, Oral Argument file, or search results.

We already have a page at: https://www.courtlistener.com/profile/api/ that just tells users their API key. We should just add a checkbox there that enables this. I know I'd love this.

**Comment by blu3r4d0n** [comment]
do you mind if I work on this? just want to make sure this is something that's still desired and figured it'd be an easy first contribution.

**Comment by mlissner** [comment]
Yep, this is definitely still desired. A few thoughts:

1. I'd start with a setting in the user's profile, as mentioned above, to turn this on or off. It should default to off, but it'd be neat if it turned on automatically. That's probably hard, but we could mention it in the API documentation too.

2. There's an example of this feature in place already on the financial disclosure [pages](https://www.courtlistener.com/person/8894/disclosure/30463/robert-e-grossman/). It's just there by default, and probably confuses people: 

    ![Image](https://github.com/user-attachments/assets/fa83ff34-631f-4c8b-b359-b72a10ac261d)

    (The little `</>` icons are what I'm thinking of.)

3. There's a revamp of the case law pages that will land in a few days, hopefully, which should make this work nicely. I'd avoid tweaking those pages before then. 

4. We have sub-issues turned on for this repo. It's a new Github feature. If you're able, I'd love it if you created sub-issues for tackling this and then got them done one PR at a time.

This is great. It'll definitely help people figure out the APIs, thank you!

**Comment by blu3r4d0n** [comment]
One  further clarification. For the existing button on the financial disclosures, would you like that to be toggled with the setting mentioned above?

**Comment by mlissner** [comment]
> One further clarification. For the existing button on the financial disclosures, would you like that to be toggled with the setting mentioned above?

Yes? Maybe? Those pages are generated with React, so they're a pain to deal with. Honestly, if you're interested, converting those pages to boring Django templates could be step one, and then including their `</>` icons would be much easier. Of course, if that feels like a pain or out of scope, then I'd say don't bother and just leave 'em as they are.

**Comment by blu3r4d0n** [comment]
Yeah, see I was expecting those to be Django templates until I looked at the code base last night and and was surprised those pages were react. If you're fine with that being a sub-issue of this issue I'll happily do it. Will be easier to maintain later when something needs updating if it's all just Django templates.

**Comment by mlissner** [comment]
Wonderful, yeah, if you want to make it a sub-issue and go for it, that'd be great. The one thing in there that might benefit from react is the typeahead dropdown when you look up judges, but I think we can keep that while redoing the rest.

Thank you!
